Documentation/Input/CreateDataProvider.md

```diff
@@ -0,0 +1,161 @@
+# Creating an input system data provider
+
+The Mixed Reality Toolkit input  system is a flexible and extensible system for enabling input device support.
+It is the input system data provider (also called device manager) that is largely responsible for providing
+support for new hardware platforms.
+
+This article describes how to create custom data providers for the input system. The example code shown here is
+from the [`WidnowsMixedRealityDeviceManager`](xref:Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityDeviceManager).
+
+
+## Implement the device manager
+
+> The complete code for the examples in this section are from the MixedRealityToolkit.Providers\WindowsMixedReality\WindowsMixedRealityDeviceManager.cs file.
+
+### Specify interface and/or base class inheritance
+
+All input system data providers must implement the [`IMixedRealityInputDeviceManager`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputDeviceManager)
+interface, which specifies the minimium functionality required by the input system. The MRTK foundation includes the [`BaseInputDeviceManager`](xref:Microsoft.MixedReality.Toolkit.Input.BaseInputDeviceManager)
+class which provides a default implementation of this required functionality.
+
+Using the example of the [`WindowsMixedRealityDeviceManager`](xref:Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input.WindowsMixedRealityDeviceManager), the following code specifies
+that the class uses the [`BaseInputDeviceManager`](xref:Microsoft.MixedReality.Toolkit.Input.BaseInputDeviceManager) and implments the
+[`IMixedRealityCapabilityCheck`](xref:Microsoft.MixedReality.Toolkit.IMixedRealityCapabilityCheck) interface.
+
+``` c#
+public class WindowsMixedRealityDeviceManager : BaseInputDeviceManager, IMixedRealityCapabilityCheck
+{ }
+```
+
+> `IMixedRealityCapabilityCheck` is used by the `WindowsMixedRealityDeviceManager` to indicate that it provides support for a set of input capabilities, specifically; articulated hands,
+gaze-gesture-voice hands and motion controllers.
```

Comment:
happy to add an example of specifying the capabilities supported.

Reply:
Wanted to add that, thinking about this the other way around, if I'm only going to provide controllers, do I need this interface since it's not a "set of input capabilities"? If not, how is it providing controller support?